### PR TITLE
feat: Support saving jsonb auditAttributes

### PR DESCRIPTION
**Pull request by vietnguyen**
https://jira.dhis2.org/browse/DHIS2-8123

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/audit/AuditAttribute.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
```
Comment (luciano-fiandesio) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/audit/AuditAttribute.java]: Please, add some Javadoc to explain what is annotation is used for

File: dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/legacy/DefaultAuditObjectFactory.java [hunk 2]
```diff
@@ -46,9 +59,24 @@
...
+        this.cacheProvider = cacheProvider;
+    }
+
+    @PostConstruct
+    public void initCache()
+    {
+        cachedAuditAttributeFields = cacheProvider.newCacheBuilder( Field.class, Method.class )
+            .forRegion( "auditAttributeFields" )
+            .withInitialCapacity( 100 )
+            .withMaximumSize( 500 )
```
Comment (mortenoh) [inline dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/legacy/DefaultAuditObjectFactory.java]: We should probably keep this under watch and make sure we don't kick any fields out... we have quite a lot of classes, and while I'm not sure how many places we will add this attribute, we should never be in a situation where some fields are kicked out of the cache

File: dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/legacy/DefaultAuditObjectFactory.java [hunk 3]
```diff
@@ -67,6 +95,38 @@ public Object create( AuditScope auditScope, AuditType auditType, Object object,
...
+        } );
+
+        return auditAttributes;
+    }
+
+    private Map<Field, Method> getAuditAttributeFields( Class<?> auditClass )
+    {
+        Optional<Map<Field, Method>> listFields = cachedAuditAttributeFields.get( auditClass.getName(), a -> AnnotationUtils.getAnnotatedFields( auditClass, AuditAttribute.class ) );
+
+        return listFields.orElse( new HashMap<>() );
```
Comment (mortenoh) [inline dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/legacy/DefaultAuditObjectFactory.java]: Could probably be collapsed to a single return? no need to split up here

File: dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/legacy/DefaultAuditObjectFactory.java (hunk 2 above)
Comment (vietnguyen) [reply dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/legacy/DefaultAuditObjectFactory.java]: increased it to 500-1000

**Pull request merged**